app/controllers/StudyController.java

```diff
@@ -28,6 +37,20 @@ public Result getStudyForResearcher() throws Exception {
         getAuthenticatedResearchOrAdminSession(study);
         return okResult(new StudyInfo(study));
     }
+    
+    public Result sendStudyParticipantsRoster() throws Exception {
+        Study study = studyService.getStudyByHostname(getHostname());
+        // Researchers only, administrators cannot get this list so easily
+        getAuthenticatedResearchSession(study);
+        
+        // One thing we can do here is verify that an email address has been set for the email
+        // that is about to be sent.
+        if (StringUtils.isBlank(study.getConsentNotificationEmail())) {
+            throw new BridgeServiceException("No consent notification contact email exists for this study. You must update the study with this contact before retrieving a roster of study participants.");
+        }
```

Comment:
Should this check be in the controller or in the service? I feel it would be better to have the controllers be minimal and have all the validation in the service.